cluster-api-resource-monitoring/README.md

```diff
@@ -0,0 +1,327 @@
+        group: cluster.x-k8s.io
+        kind: Machine
+        version: v1beta1
+      metrics: 
+        - each: 
+            path: 
+              - status
+              - phase
+          help: "machine phase"
+          name: phase
+```
+
+will lead to following error:
+
+```
+E0607 09:12:00.168168  126676 registry_factory.go:469] "kube_cluster_x-k8s_io_v1beta1_Machine_phase" err="[status,phase]: []: strconv.ParseFloat: parsing \"Running\": invalid syntax"
+```
+
+This behavior got already addressed to Cluster API community in [issue #6458](https://github.com/kubernetes-sigs/cluster-api/issues/6458#issuecomment-1148293873). The current discussion together with the KSM folks is one in [this slack thread](https://kubernetes.slack.com/archives/CJJ529RUY/p1654593034854759).
+
+With the assumption, that `kube-state-metrics` accept patches which might make CASM obsolete, this section will cover the integration of Cluster API infrastructure provider specific `CustomResourceStateMetrics` configuration.
+
+### Variant 1: specific `kube-state-metrics` configuration for the already existing `kube-state-metrics-app`
+
+As `kube-state-metrics` is already deployed in each cluster (no matter if management- or workload cluster), a condition has to be implemented to detect, if the current KSM instance is running on a management- or a workload cluster. With this condition in place, we can provide a CAPI generic `CustomResourceStateMetrics` configuration plus one `CustomResourceStateMetrics` configuration per infrastructure provider.
+
+#### Pros
+
+- CAPI generic `CustomResourceStateMetrics` configuration could be easily re-used on different providers
+- KSM-App already exists
+
+#### Cons
+
+- Scope of KSM between management- and workload cluster differs
+  - resource requirements
+  - permissions
+- Changes in KSM-App affect all clusters, even if changes are only done for management clusters (e.g. version bump)
+
+### Variant 2: dedicated `kube-state-metrics` on management cluster
+
+As `kube-state-metrics` is already deployed in each cluster with a well defined scope, it's also possible to create a dedicated CAPI scoped KSM-App to only monitor Cluster API plus infrastructure specific custom resources.
+
+#### Variant 2.1: dedicated `kube-state-metrics` instance on management cluster for all CAPI providers
+
+One generic KSM instance which bundles all the Cluster API specific `CustomResourceStateMetrics` configuration plus infrastructure specific specific `CustomResourceStateMetrics` configuration.
+
+##### Pros
+
+- Scope of this KSM instance is clearly focused on everything CAPI
+- Any further management cluster CR specific monitoring clearly goes there
+
+##### Cons
+
+- Cluster API versions must be reflect in the `CustomResourceStateMetrics` configuration. We will potentially diverge on the used Cluster API versions per infrastructure provider.
+
+#### Variant 2.2: dedicated `kube-state-metrics` instance on management cluster per CAPI providers
```

Comment:
For this I understand we could put a bunch of `{{ if eq .Values.provider "openstack" }}` in the helm chart and that would do right?

Reply:
With Variant 1 and Variant 2.1 we will have a lot of `{{ if eq .Values.provider "openstack" }}` in the repos (and therefore we have to decide how to go with corresponding APP releases - as we could break other providers monitoring)